website/docs/mocking.md

```diff
@@ -125,131 +136,189 @@ const schemaWithMocks = addMocksToSchema({
+  schema,
+  store,
+  resolvers: (store) => ({
+    Query {
+      userById(_, { id }) => store.get('User', id),
+    }
+  })
+});
+```
 
-type Distributor {
-  id: Int
-  name: String
-}
+Note that, by default, the `id` or `_id` field will be used as storage key and the store will make sure the storage key and the field value are equal. You can change the key field using the option `typePolicies`.
 
-type Product {
-  id: Int
-  name: String
-}
+### Mocking a pagination
 
-interface List {
-  amount: Int
-  offset: Int
-  total: Int
-  remaining: Int
-}
+The idea is that the store contains the full list, as field value, and that the resolver queries the store and slice the results:
 
-type DistributorList implements List {
-  amount: Int
-  offset: Int
-  total: Int
-  remaining: Int
-  items: [Distributor]
+```ts
+const typeDefs = `
+type User {
+  id: Id!
+  name: String!
+  friends(offset: Int!, limit: Int!): [User!]!
 }
-
-type ProductList implements List {
-  amount: Int
-  offset: Int
-  total: Int
-  remaining: Int
-  items: [Product]
+type Query {
+  me: User!
 }
 `
-
-const resolvers = {
-  List: {
-    __resolveType(data) {
-      return data.typename // typename property must be set by your mock functions
+const schema = makeExecutableSchema({ typeDefs: schemaString });
+const schemaWithMocks = addMocksToSchema({
+  schema,
+  store,
+  resolvers: (store) => ({
+    User: {
+      // `addMocksToSchema` resolver will pass a `Ref` as `parent`
+      // it contains a key to the `User` we are dealing with
+      friends: (userRef, { offset, limit }) => {
+        // this will generate and store a list of `Ref`s to some `User`s
+        // next time we go thru this resolver (with same parent), the list
+        // will be the same
+        const fullList = store.get(userRef, 'friends') as Ref[];
+
+        // actually apply pagination slicing
+        return fullList.slice(offset, offset + limit)
+      }
     }
-  }
-}
+  })
+});
+```
 
-const schema = makeExecutableSchema({
-  typeDefs,
-  resolvers
-})
+#### Relay-style pagination
+The principles stay the same than for basic pagination:
 
+```ts
+const typeDefs = `
+type User {
+  id: Id!
+  name: String!
+  friends(offset: Int!, limit: Int!): FriendsConnection;
+}
+type FriendsConnection {
+  totalCount: Int!
+  edges: [FriendConnectionEdge!]!
+}
+type FriendsConnectionEdge {
+  node: User!
+}
+type Query {
+  me: User!
+}
+`
+const schema = makeExecutableSchema({ typeDefs: schemaString });
+const store = createMockStore({ schema });
 const schemaWithMocks = addMocksToSchema({
   schema,
-  mocks,
-  preserveResolvers: true
-})
+  store,
+  resolvers: (store) => ({
+    User: {
+      friends: (userRef, { offset, limit }) => {
+
+        const connectionRef = store.get(userRef, 'friends', 'edges');
+
+        return {
+          totalCount: edgesFullList.length,
+          edges: edgesFullList.slice(offset, offset + limit)
+      }
+    }
+  })
+});
 ```
```

Comment:
This is awesome stuff. Huge improvement over current mocking package, which I actually don’t find to be more useful than just simple resolvers plugged into an executable schema. 

This looks like a great addition to v8